### ISSUE TITLE
Slab Frame crash when using a block in it

### ISSUE THREAD

**Issue by W1nter-isHere** [PianoManu/BlockCarpentry#46]
Crash report: https://pastebin.com/i4fYGCgV
Forge version: 36.2.0
Minecraft version: 1.16.5
Mod version: 0.3.8

Thank you!

**Comment by PianoManu** [comment]
Thanks for reporting. Seems like I didn't test the "double slab" feature enough...
Regards, Manu

**Comment by PianoManu** [comment]
Should be fixed with [0.3.9](https://www.curseforge.com/minecraft/mc-mods/blockcarpentry/files/3430443)...

**Comment by wchen1990** [comment]
This is not fixed in 0.3.11.

Since it doesn't seem the repo is up to date, I looked at the decompiled jar of 0.3.11 and I still see the `checkVisibility` method that references a method `func_200122_a` that is marked as client side only using `@OnlyIn(Dist.CLIENT)` for both `SixWaySlabFrameBlock` and `FrameBlock`.

**Comment by Seika85** [comment]
```
[22Aug2021 13:18:21.665] [Server thread/ERROR] [net.minecraft.server.MinecraftServer/]: Encountered an unexpected exception
java.lang.NoSuchMethodError: 'boolean mod.pianomanu.blockcarpentry.block.SixWaySlabFrameBlock.func_200122_a(net.minecraft.block.BlockState, net.minecraft.block.BlockState, net.minecraft.util.Direction)'
	at mod.pianomanu.blockcarpentry.block.SixWaySlabFrameBlock.checkForVisibility(SixWaySlabFrameBlock.java:298) ~[blockcarpentry:1.16-0.3.11]
	at mod.pianomanu.blockcarpentry.block.SixWaySlabFrameBlock.func_225533_a_(SixWaySlabFrameBlock.java:180) ~[blockcarpentry:1.16-0.3.11]
	at net.minecraft.block.AbstractBlock$AbstractBlockState.func_227031_a_(AbstractBlock.java:640) ~[?:?]
	at net.minecraft.server.management.PlayerInteractionManager.func_219441_a(PlayerInteractionManager.java:338) ~[?:?]
	at net.minecraft.network.play.ServerPlayNetHandler.func_184337_a(ServerPlayNetHandler.java:958) ~[?:?]
	at net.minecraft.network.play.client.CPlayerTryUseItemOnBlockPacket.func_148833_a(SourceFile:36) ~[?:?]
	at net.minecraft.network.play.client.CPlayerTryUseItemOnBlockPacket.func_148833_a(SourceFile:10) ~[?:?]
	at net.minecraft.network.PacketThreadUtil.func_225383_a(SourceFile:21) ~[?:?]
	at net.minecraft.util.concurrent.TickDelayedTask.run(SourceFile:18) ~[?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213166_h(SourceFile:144) ~[?:?]
	at net.minecraft.util.concurrent.RecursiveEventLoop.func_213166_h(SourceFile:23) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_213166_h(MinecraftServer.java:734) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_213166_h(MinecraftServer.java:159) ~[?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213168_p(SourceFile:118) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_213205_aW(MinecraftServer.java:717) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_213168_p(MinecraftServer.java:711) ~[?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213160_bf(SourceFile:103) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_213202_o(MinecraftServer.java:696) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:646) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_240783_a_(MinecraftServer.java:232) ~[?:?]
	at java.lang.Thread.run(Thread.java:829) [?:?]

```[22Aug2021 13:18:21.665] [Server thread/ERROR] [net.minecraft.server.MinecraftServer/]: Encountered an unexpected exception
java.lang.NoSuchMethodError: 'boolean mod.pianomanu.blockcarpentry.block.SixWaySlabFrameBlock.func_200122_a(net.minecraft.block.BlockState, net.minecraft.block.BlockState, net.minecraft.util.Direction)'
	at mod.pianomanu.blockcarpentry.block.SixWaySlabFrameBlock.checkForVisibility(SixWaySlabFrameBlock.java:298) ~[blockcarpentry:1.16-0.3.11]
	at mod.pianomanu.blockcarpentry.block.SixWaySlabFrameBlock.func_225533_a_(SixWaySlabFrameBlock.java:180) ~[blockcarpentry:1.16-0.3.11]
	at net.minecraft.block.AbstractBlock$AbstractBlockState.func_227031_a_(AbstractBlock.java:640) ~[?:?]
	at net.minecraft.server.management.PlayerInteractionManager.func_219441_a(PlayerInteractionManager.java:338) ~[?:?]
	at net.minecraft.network.play.ServerPlayNetHandler.func_184337_a(ServerPlayNetHandler.java:958) ~[?:?]
	at net.minecraft.network.play.client.CPlayerTryUseItemOnBlockPacket.func_148833_a(SourceFile:36) ~[?:?]
	at net.minecraft.network.play.client.CPlayerTryUseItemOnBlockPacket.func_148833_a(SourceFile:10) ~[?:?]
	at net.minecraft.network.PacketThreadUtil.func_225383_a(SourceFile:21) ~[?:?]
	at net.minecraft.util.concurrent.TickDelayedTask.run(SourceFile:18) ~[?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213166_h(SourceFile:144) ~[?:?]
	at net.minecraft.util.concurrent.RecursiveEventLoop.func_213166_h(SourceFile:23) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_213166_h(MinecraftServer.java:734) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_213166_h(MinecraftServer.java:159) ~[?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213168_p(SourceFile:118) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_213205_aW(MinecraftServer.java:717) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_213168_p(MinecraftServer.java:711) ~[?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213160_bf(SourceFile:103) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_213202_o(MinecraftServer.java:696) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:646) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_240783_a_(MinecraftServer.java:232) ~[?:?]
	at java.lang.Thread.run(Thread.java:829) [?:?]

**Comment by PianoManu** [comment]
Alright, I'm just plain stupid. I was in a hurry when I first tried to fix this issue and somehow got the client and dedicated server confused, and I didn't test it afterwards (because I thought I fixed it properly). So I ended up rendering the slab only on the server side... Welp...

Now I tested it completely on my test server and it works. It'll be fixed in the next update. Thanks for reporting!
Regards, Manu
